### PR TITLE
Kubernetes ingress provider to search via all endpoints

### PR DESCRIPTION
**Pull request by martinvizvary**
…bsets was searched

### What does this PR do?

Fix kubernetes ingress provider to return error only if slice endpoints.Subsets was searched

### Motivation

fix https://github.com/traefik/traefik/issues/7991

### More

- [ ] Added/updated tests
- [ ] Added/updated documentation

### CODE REVIEW

File: pkg/provider/kubernetes/ingress/kubernetes.go [hunk 1]
```diff
@@ -525,31 +525,34 @@ func loadService(client Client, namespace string, backend networkingv1beta1.Ingr
...
 			}
 		}
 
-		if port == 0 {
+		if port == 0 && i == endpointsLen-1 {
 			return nil, errors.New("cannot define a port")
 		}
 
-		protocol := getProtocol(portSpec, portName, svcConfig)
+		if port != 0 {
```
Comment (dtomcej) [inline pkg/provider/kubernetes/ingress/kubernetes.go:546]:
```suggestion
		if port == 0 {
		continue
		}
```

This prevents an additional level indent, and fails fast.

**Review by jbdoumenjou — APPROVED** [review]
LGTM :+1:

**Review by dtomcej — APPROVED** [review]
LGTM
:shipit:

**Comment by martinvizvary** [comment]
hello, do you have any idea when will be the fix merged and released?

**Comment by SantoDE** [comment]
@martinvizvary soon! We plan to include this fix in our next releases

File: pkg/provider/kubernetes/ingress/fixtures/Ingress-with-port-invalid-for-one-service_ingress.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+kind: Ingress
+apiVersion: networking.k8s.io/v1beta1
```
Comment (SantoDE) [inline pkg/provider/kubernetes/ingress/fixtures/Ingress-with-port-invalid-for-one-service_ingress.yml:2]: Can we use the `networking.k8s.io/v1` version please?

**Review by SantoDE — CHANGES_REQUESTED** [review]
Just one nitpick.

Remaining things are good

File: pkg/provider/kubernetes/ingress/fixtures/Ingress-with-port-invalid-for-one-service_ingress.yml (hunk 2 above)
Comment (martinvizvary) [reply pkg/provider/kubernetes/ingress/fixtures/Ingress-with-port-invalid-for-one-service_ingress.yml:2]: could we please skip this change if it is just a nitpick? I took a quick look on the requested change and it looks like the whole testing is not ready for `networking/v1`...

Comment (SantoDE) [reply pkg/provider/kubernetes/ingress/fixtures/Ingress-with-port-invalid-for-one-service_ingress.yml:2]: Okay, in that case, lets skip it :)

**Review by SantoDE — APPROVED** [review]
LGTM 👼

**Pull request merged**